var/spack/repos/builtin/packages/yaksa/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Yaksa(AutotoolsPackage):
+    """Yaksa is a high-performance datatype engine for expressing,
+    managing and manipulating data present in noncontiguous memory
+    regions. It provides portable abstractions for structured
+    noncontiguous data layouts that are much more comprehensive compared
+    with traditional I/O vectors.
+
+    Yaksa imitates parts of the MPI Datatype system, but adds additional
+    functionality that would allow it to be used independent of MPI. It
+    provides routines for packing/unpacking, creating I/O vectors (array
+    of contiguous segments) and flattening/unflattening datatypes into
+    process-portable formats.
+
+    Yaksa's backend includes support for CPUs as well as different
+    GPUs."""
+
+    homepage = "https://www.yaksa.org"
+    url      = "https://github.com/pmodels/yaksa/archive/refs/tags/v0.2.tar.gz"
+    maintainers = ['raffenet', 'yfguo', 'hzhou']
+
+    version('0.2', sha256='9401cb6153dc8c34ddb9781bbabd418fd26b0a27b5da3294ecc21af7be9c86f2')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    variant('cuda', default=False, description="CUDA backend support")
+    depends_on('cuda', when='+cuda')
```

Comment:
Is it worth inheriting from `CudaPackage`?

Reply:
Perhaps? Note that I also want to add Level Zero and HIP variants for Intel and AMD GPU support. Not sure if that influences the choice.

Reply:
The main advantage of inheriting from CudaPackage is getting conflicts for gcc and clang for concretization, that you otherwise get at compile time when including `cuda_runtime.h`: `error -- unsupported GNU version! gcc ... and up are not supported!`

Reply:
I added CudaPackage as suggested and also added a HIP variant. Since its not immediately clear how add a Level Zero dependency, I'm going to save that for future work. Removing the WIP tag, this should be OK to review again.